bug(backend): create missing payments.ts route file — backend crashes on startup

## Bug Report

**File:** `backend/src/routes/payments.ts` (missing)

### Description
`index.ts` imports `paymentsRouter` from `'./routes/payments.js'` but this file does not exist. The backend fails to start with `Error: Cannot find module './routes/payments.js'`.

### Steps to Reproduce
1. `cd backend && npm run dev`
2. Server crashes immediately: `Cannot find module './routes/payments.js'`

### Implementation Guide

Create `backend/src/routes/payments.ts`:

    import { Router } from 'express';
    import * as StellarSdk from '@stellar/stellar-sdk';
    import { stellarService } from '../lib/stellar.js';
    import { asyncHandler } from '../lib/asyncHandler.js';
    import { z } from 'zod';

    export const paymentsRouter = Router();

    const PaymentSchema = z.object({
      meter_id: z.string().min(1).max(9),
      payer:    z.string().length(56),
      amount:   z.number().int().positive(),
      plan:     z.enum(['Daily', 'Weekly', 'UsageBased']),
    });

    paymentsRouter.post('/', asyncHandler(async (req, res) => {
      const parsed = PaymentSchema.safeParse(req.body);
      if (!parsed.success) return res.status(400).json({ errors: parsed.error.flatten().fieldErrors });
      const { meter_id, payer, amount, plan } = parsed.data;
      const hash = await stellarService.invoke('make_payment', [
        StellarSdk.nativeToScVal(meter_id, { type: 'symbol' }),
        StellarSdk.nativeToScVal(payer,    { type: 'address' }),
        StellarSdk.nativeToScVal(BigInt(amount), { type: 'i128' }),
        StellarSdk.nativeToScVal(plan,     { type: 'symbol' }),
      ]);
      res.json({ hash });
    }));

### Definition of Done
- [ ] `backend/src/routes/payments.ts` created and exports `paymentsRouter`
- [ ] `POST /api/payments` validates all fields with Zod
- [ ] Returns `{ hash }` on success
- [ ] Returns `400` with field errors on invalid input
- [ ] Backend starts without module-not-found error

--- a/backend/src/routes/meters.ts
+++ b/backend/src/routes/meters.ts
@@ -14,6 +14,42 @@ const BALANCE_CACHE_TTL_MS = 5_000; // 5-second cache to reduce RPC load
 export function createMeterRouter(stellar: StellarService) {
   const meterRouter = Router();
 
+  /**
+   * GET /api/meters?page=1&pageSize=25 — list all meters with pagination
+   *
+   * Registered BEFORE /:id so the literal string "meters" is never matched
+   * as a meter ID parameter.
+   *
+   * Fixes #268.
+   */
+  meterRouter.get(
+    "/",
+    asyncHandler(async (req, res) => {
+      const page = Math.max(1, Number(req.query.page ?? 1) || 1);
+      const pageSize = Math.min(
+        100,
+        Math.max(1, Number(req.query.pageSize ?? 25) || 25),
+      );
+
+      const result = await stellar.query("get_all_meters", []);
+      const allMeters = (StellarSdk.scValToNative(result) as any[]) ?? [];
+
+      const total = allMeters.length;
+      const start = (page - 1) * pageSize;
+      const meters = allMeters.slice(start, start + pageSize);
+
+      res.json({
+        meters,
+        pagination: {
+          page,
+          pageSize,
+          total,
+          pages: Math.ceil(total / pageSize),
+        },
+      });
+    }),
+  );
+
   /** GET /api/meters/export?format=csv|json — download all meter data */
   meterRouter.get(
     "/export",

--- a/backend/src/routes/webhooks.ts
+++ b/backend/src/routes/webhooks.ts
@@ -1,7 +1,7 @@
 import { Router } from "express";
 import * as crypto from "crypto";
 import * as StellarSdk from "@stellar/stellar-sdk";
-import { adminInvoke, contractQuery } from "../lib/stellar.js";
+import { adminInvoke } from "../lib/stellar.js";
 import { asyncHandler } from "../lib/asyncHandler.js";
 import { validateRequest } from "../lib/validation.js";
 import { logger } from "../lib/logger.js";

